tests/unit/stanmath/StanPerceptronTest_unittest.cc

```diff
@@ -0,0 +1,73 @@
+#include <gtest/gtest.h>
+#include <stan/math.hpp>
```

Comment:
i'm not so sure if this is the best way... namely that you are including stan like this.
meaning what if for some reason `stan/math.hpp` is in the include path already... probably the `-isystem` inclusion will take care of this, but maybe it'd be better that we explicitly specify in the include path that it's shogun's third-party include path... on the other hand probably this will actually break everything else within stan's headers.... 

it's more important actually that stan usage (or a model that is actually using stan) is being tested from a libshogun example. namely that the cmake target flags are set correctly when shogun is being used from a third-party lib, or just a simple CLI. note that it requires that either you:
 * either include stan only in cpp files of shogun, or
 * you expose the include flags in ShogunTarget.cmake; currently it's `PUBLIC SYSTEM` which should (? does it really? should be tested) reflect the correct exposure of include flags.... but i'm not so sure if we want to expose stan to the public, i would presume it's safe to think that we only use stan in the implementation of the models.